xchange-lgo/src/main/java/info/bitrich/xchangestream/lgo/LgoStreamingTradeService.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    /**
+     * Get ack for your placed orders. "received" events indicate the orderId associated to your
+     * order, if you set a reference on order placement you will have it in the event. "failed" events
+     * indicate that the order could not be read or was invalid and not added to a batch.
+     */
+    public Observable<LgoOrderEvent> getReceivedOrderEvents() {
+        if (afrSubscription == null) {
+            createAfrSubscription();
+        }
+        return afrSubscription.share();
+    }
+
+    private void createAfrSubscription() {
+        final ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+        afrSubscription = streamingService
+                .subscribeChannel("afr")
+                .map(s -> mapper.readValue(s.toString(), LgoAckUpdate.class))
+                .map(LgoAckUpdate::getData)
+                .flatMap(Observable::fromIterable)
+                .map(LgoAdapter::adaptReceivedEvent);
+    }
+
+    /**
+     * Get all events of your orders happened during batch execution, for a currency pair. "pending"
+     * events indicate that the order was added to a batch and received by the execution engine.
+     * "invalid" events indicate that the order was not suitable for execution. "match" events
+     * indicate that the order did match against another order. "open" events indicate that the order
+     * entered the order book. "done" events indicate that the order was filled, canceled or
+     * rejected.
+     */
+    public Observable<LgoOrderEvent> getBatchOrderEvents(CurrencyPair currencyPair) {
+        return getOrderUpdates(currencyPair)
+                .map(LgoBatchUpdate::getEvents)
+                .flatMap(Observable::fromIterable);
+    }
+
+    /**
+     * Place a market order
+     *
+     * @return the order reference
+     */
+    public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
+        Long ref = nonceFactory.createValue();
+        LgoPlaceOrder lgoOrder = LgoAdapters.adaptMarketOrder(marketOrder);
+        return placeOrder(ref, lgoOrder);
+    }
+
+    /**
+     * Place a limit order
+     *
+     * @return the order reference
+     */
+    public String placeLimitOrder(LimitOrder limitOrder) throws IOException {
+        Long ref = nonceFactory.createValue();
+        LgoPlaceOrder lgoOrder = LgoAdapters.adaptLimitOrder(limitOrder);
+        return placeOrder(ref, lgoOrder);
+    }
+
+    /**
+     * Place a cancel order
+     *
+     * @return true
+     */
+    public boolean cancelOrder(String orderId) throws IOException {
+        Long ref = nonceFactory.createValue();
+        LgoPlaceCancelOrder lgoOrder = new LgoPlaceCancelOrder(ref, orderId, new Date().toInstant());
+        placeOrder(ref, lgoOrder);
+        return true;
+    }
+
+    private String placeOrder(Long ref, LgoPlaceOrder lgoOrder) throws JsonProcessingException {
```

Comment:
This is interesting! Not seen exchanges which permit placing orders over the websocket before.